### PR TITLE
Sharing: Update to FoldableCard Pattern

### PR DESCRIPTION
**Pull request by alternatekev**
This PR updates Sharing to the `FoldableCard` components. Currently in production, Sharing looks like this:
screen shot 2016-01-14 at 3 03 17 pm

When updating to FoldableCards, this PR looks like this:

screen shot 2016-01-26 at 4 35 29 pm

This also makes use of the new `SocialLogo` component introduced by @enejb. Since Twitter doesn't have a box-surrounded version of the icon, I went with mostly non-boxed icons here. We can easily change this to include boxed icons for every logo, but it'd require adding one for Twitter to the `SocialLogo` component.

Pinging @johnHackworth, @rickybanister and @mtias for reviews.

### CODE REVIEW

**Comment by rickybanister** [comment]
The borderless buttons look just fine to me.

Do you think it would be possible (later on, different PR) to switch to foldable card for these?

**Comment by alternatekev** [comment]
@rickybanister Yep, that was going to be my next step. Do we even need them foldable if there's no connection? The preview screenshots don't seem to add much.

**Comment by kellychoffman** [comment]
What is the reasoning behind the change?

**Comment by kellychoffman** [comment]
>  switch to foldable card

> Do we even need them foldable if there's no connection? The preview screenshots don't seem to add much.

We are planning on updating the page to use more components, such as the foldable card. The preview screenshots are important in that they tell a new user what they can get out of connecting without sending them to a support doc.

**Comment by alternatekev** [comment]
@kellychoffman Updated to remove the icon:

screen shot 2015-12-11 at 4 04 40 pm

**Comment by kellychoffman** [comment]
It looks a bit off when mixed with services that aren't connected. Maybe because it isn't aligned.
screen shot 2015-12-13 at 10 06 27 am

Do you think `Disconnect` looks like a link?

**Comment by rickybanister** [comment]
@kellychoffman would it be inappropriate for it to be red? Is disconnecting a sharing service a 'scary' thing? It would then be inline with many other 'remove' and 'delete' items.

File: client/my-sites/sharing/connections/service-action.jsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React from "react";
```
Comment (gwwar) [inline client/my-sites/sharing/connections/service-action.jsx]: Single quotes please. :)

File: client/my-sites/sharing/connections/service-action.jsx [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 
 /**
  * Internal dependencies
  */
-var serviceConnections = require( './service-connections' );
+import serviceConnections from "./service-connections";
+import Gridicon from "components/gridicon";
+import Button from "components/button";
+import classNames from "classnames";
```
Comment (gwwar) [inline client/my-sites/sharing/connections/service-action.jsx]: Single quotes please. :)

File: client/my-sites/sharing/connections/service-action.jsx [hunk 3]
```diff
@@ -38,27 +41,32 @@ module.exports = React.createClass( {
 	},
 
 	render: function() {
-		var classes = [ 'sharing-service-action', 'button' ],
-			isPending, removableConnections, label;
+		var classes = [ 'sharing-service-action' ],
```
Comment (gwwar) [inline client/my-sites/sharing/connections/service-action.jsx]: We can use `let` here instead of `var`

**Comment by gwwar** [comment]
Besides some minor changes the code looks good :+1: This branch needs to rebased.

> Do you think Disconnect looks like a link?

It does. Having this large of a click target for something that looks like a link, is a bit odd to me.

clickable

**Comment by alternatekev** [comment]
So.. I sort of revamped the whole thing and came up with this, which should solve most people's issues with this page for a while. We could probably get rid of the icons though:

screen shot 2015-12-16 at 2 30 04 pm

**Comment by alternatekev** [comment]
Here's how it looks icon-less:

screen shot 2015-12-16 at 2 42 07 pm

**Comment by alternatekev** [comment]
The above changes would work well also on #1477

**Comment by rickybanister** [comment]
This isn't my game, but icon-less is better. Ironically we're nearly back to what it was before. Just with the new foldable card style.

**Comment by rralian** [comment]
@alternatekev is [the mock here](https://github.com/Automattic/wp-calypso/pull/1474#issuecomment-165258107) just a visual mockup or was it a screenshot? The alignment of the icons is inconsistent. Otherwise I don't have a dog in the icon/iconless fight.

**Comment by mtias** [comment]
I like the compact buttons since they weight less in a page otherwise full of buttons.

**Comment by alternatekev** [comment]
@rralian This was a screenshot, and you're right about the icon alignment. It was looking super weird to me too, which should probably be its own PR/issue.

**Comment by alternatekev** [comment]
> This isn't my game, but icon-less is better. Ironically we're nearly back to what it was before. Just with the new foldable card style.

Yeah, I got myself to a place where I didn't think the borderless was working here in practice. To be fair, we're using the compact buttons which I think make a ton of difference, especially when taking into account the foldable card updates. It feels correct to me, which isn't to say it 100% is.

**Comment by alternatekev** [comment]
@kellychoffman what do you think about moving eventbrite up to the main area and treating it the same? does it need to be in its own group down there:

screen shot 2015-12-17 at 11 24 07 am

**Comment by kellychoffman** [comment]
I was out for the past two days... Catching up. 

> icon-less is better. Ironically we're nearly back to what it was before.

I like that we are back to the buttons, but I'm not convinced compact is better than the original buttons. For a new user, Connecting is the primary (only) action for this page. Also, I agree about the icons - I'm not sure they are adding anything here.

> what do you think about moving eventbrite up to the main area and treating it the same? does it need to be in its own group down there:

There's going to be other services down there. Also Eventbrite isn't publicizing, so I'd rather not group it up with those services.

**Comment by jkudish** [comment]
I agree with @kellychoffman -- grouping the services would be pretty confusing for users. There will eventually be other services there too... What Eventbrite does, and how it works is very different from Publicize. I've even advocated for a different tab for it in the past, though I understand why we didn't do that (at least for now).

**Comment by folletto** [comment]
I give it a run on Chrome, Firefox, Safari — feels it's working really well to me. :)

One minor thing I noticed, is that the animation on the chevron has disappeared. But I guess it's because this is using the same FoldableCard pattern, but not the FoldableCard component, right?

**Comment by alternatekev** [comment]
The version of this PR that you can test is using faked FoldableCards, yse. I’ve got commits waiting to be pushed that replace the whole thing with actual FoldableCards.

**Comment by folletto** [comment]
Ok, cool! :)

**Comment by alternatekev** [comment]
This one's ready for review @folletto @kellychoffman @mtias.

It doesn't include the border since that's a part of the FoldableCard update which you can see a preview of here: https://github.com/Automattic/wp-calypso/pull/2390#issuecomment-171790649

**Comment by rickybanister** [comment]
What does this look like now? Could update the initial description a bit since things have changed quite a lot in a month?

**Comment by alternatekev** [comment]
@rickybanister yep, updating now.

**Comment by drw158** [comment]
If using the social icons feels misaligned, please note it in the social icons repo so we can get it fixed.

**Comment by folletto** [comment]
Tested:

* Open/Close
* Connecting and cancelling
* Active connections

On: Chrome, Firefox, Safari, IE11, Edge. All works. :+1:

**Comment by enejb** [comment]
From the looks in the code it looks like you woun't need .svg files anymore.

**Comment by alternatekev** [comment]
@enejb SVGs have been removed.

**Comment by enejb** [comment]
It looks like there might be a problem here with smaller screens. 
screen shot 2016-01-25 at 12 07 36

I am not sure if you want to fix this in this PR or in a different one. 

Other then the ^ it works pretty well.

File: client/my-sites/sharing/connections/service-placeholder.jsx [hunk 4]
```diff
@@ -3,6 +3,8 @@
  */
 var React = require( 'react' );
 
+var Button = require( 'components/Button' );
```
Comment (enejb) [inline client/my-sites/sharing/connections/service-placeholder.jsx]: We need to use the lowercase Button here here. `require( 'components/button' );` 
react is throwing me some warnings. 
screen shot 2016-01-25 at 12 23 59

**Comment by alternatekev** [comment]
Pinging @enejb for another review.

**Comment by enejb** [comment]
I am still able to see ^ error. 
screen shot 2016-01-26 at 16 49 41

**Comment by alternatekev** [comment]
@enejb that's weird, it should be looking like this now:

screen shot 2016-01-26 at 7 58 34 pm

**Comment by enejb** [comment]
Maybe I am testing a width that is too narrow?
Will try to make sure that my CSS is also the latest version just in case.

**Comment by folletto** [comment]
I get the same issue in Chrome and Safari (iPhone 5 emulation): 

screen shot 2016-01-27 at 10 57 07

Does **not** happen in Firefox.

Starts happening below ~350px of width.

Maybe we should refactor that part of the component to use `flex` which will make the component far more solid?

**Comment by alternatekev** [comment]
@folletto 
> Maybe we should refactor that part of the component to use flex which will make the component far more solid?

I'd be happy to do this. Sharing & Connected apps both use custom CSS to do their icon/layout work which get in the way of the component's native rendering a bit. Flexbox would help out the FoldableCard `header` and `summary` props considerably.

**Comment by alternatekev** [comment]
@folletto @enejb I addressed the flexbox layout in #2835

**Comment by enejb** [comment]
There are still some widths that looks strange to me when I do the test the page and resize it. 
For example 
screen shot 2016-01-27 at 15 54 33

Which to me just looks odd/broken.

**Comment by folletto** [comment]
It's a bit odd, but better than live right now where the text at certain breakpoints goes below the button. :P 

We can probably fix the content with some extra rule (again `flex`?) to make that more flexible as well.

**Comment by alternatekev** [comment]
I'm going to do a new PR to adjust the flex widths at small sizes so I think this one is ready to merge?

**Comment by kellychoffman** [comment]
I think we should keep the white icons with the colored background. The updated colored icons looks too messy.

**Comment by MichaelArestad** [comment]
Just out of curiosity, why were the icons taken out of the colored backgrounds? It looked more uniform before.

**Comment by alternatekev** [comment]
The icons were changed because there's a new SocialLogo component that includes all of the icons within a roundrect except for Twitter. We could go with all roundrect backgrounds but I need one for Twitter added to the component.

**Comment by kellychoffman** [comment]
Could we use the non `roundrect` versions of all the logos instead? (I think Facebook is the only one without one) For Facebook, we could use the `roundrect` version and just make sure it matches the border radius of the others.

**Comment by alternatekev** [comment]
I really wanted to avoid doing custom CSS for all of the logos except one. If almost all have versions, it shouldn't be too hard to add just one. Also it makes the alignment of the logo easier because it's being done in SVG vs aligning it around in CSS.

**Comment by kellychoffman** [comment]
Fair. I think we should avoid making updates to the social icons until we have either all roundrect or all standalone.

**Comment by alternatekev** [comment]
By adding a roundrect version of Twitter (which I'm just about done with), then this page will have all roundrect. Except Google+ which will be a circle and Eventbrite which will have its off-kilter rectangle. Would that work?

**Comment by drw158** [comment]
> By adding a roundrect version of Twitter (which I'm just about done with), then this page will have all roundrect. Except Google+ which will be a circle and Eventbrite which will have its off-kilter rectangle. Would that work?

That works for me. I'm not opposed to including a version of the Twitter logo with a rectangle.

**Comment by alternatekev** [comment]
> Could we use the non roundrect versions of all the logos instead? (I think Facebook is the only one without one) For Facebook, we could use the roundrect version and just make sure it matches the border radius of the others.

@kellychoffman I totally misread this as you wanting to use ALL roundrects. This is true, except for LinkedIn, which also is only in a roundrect format. Using all non-roundrect except for LinkedIn and Facebook is what this PR does now.

I've already got the SocialLogos component updated with a pending PR to add the Twitter roundrect... we could see how that version stacks up against what's in the screenshot at the top and go from there.

**Comment by kellychoffman** [comment]
> Using all non-roundrect except for LinkedIn and Facebook is what this PR does now.

Right, but not with colored backgrounds as it was before. 

> I always forget about Eventbrite I guess. It seems rounrect-enough to me?

Yeah I think its fine on its own.

**Pull request merged**